manage_proposal.inc

```diff
@@ -0,0 +1,1141 @@
+{
+    if ($form_state['values']['approval'] == 2) {
+        if ($form_state['values']['message'] == '') {
+            form_set_error('message', t('Reason for disapproval could not be empty'));
+        } //$form_state['values']['message'] == ''
+    } //$form_state['values']['approval'] == 2
+}
+function esim_research_migration_proposal_approval_form_submit($form, &$form_state)
+{
+    global $user;
+    /* get current proposal */
+    $proposal_id = (int) arg(3);
+    $query = db_select('research_migration_proposal');
+    $query->fields('research_migration_proposal');
+    $query->condition('id', $proposal_id);
+    $proposal_q = $query->execute();
+    if ($proposal_q) {
+        if ($proposal_data = $proposal_q->fetchObject()) {
+            /* everything ok */
+        } //$proposal_data = $proposal_q->fetchObject()
+        else {
+            drupal_set_message(t('Invalid proposal selected. Please try again.'), 'error');
+            drupal_goto('research-migration-project/manage-proposal');
+            return;
+        }
+    } //$proposal_q
+    else {
+        drupal_set_message(t('Invalid proposal selected. Please try again.'), 'error');
+        drupal_goto('research-migration-project/manage-proposal');
+        return;
+    }
+    if ($form_state['values']['approval'] == 1) {
+        $query = "UPDATE {research_migration_proposal} SET approver_uid = :uid, approval_date = :date, approval_status = 1 WHERE id = :proposal_id";
+        $args = array(
+            ":uid" => $user->uid,
+            ":date" => time(),
+            ":proposal_id" => $proposal_id,
+        );
+        db_query($query, $args);
+        /* sending email */
+        $user_data = user_load($proposal_data->uid);
+        $email_to = $user_data->mail;
+        $from = variable_get('research_migration_from_email', '');
+        $bcc = $user->mail . ', ' . variable_get('research_migration_emails', '');
+        $cc = variable_get('research_migration_cc_emails', '');
+        $params['research_migration_proposal_approved']['proposal_id'] = $proposal_id;
+        $params['research_migration_proposal_approved']['user_id'] = $proposal_data->uid;
+        $params['research_migration_proposal_approved']['headers'] = array(
+            'From' => $from,
+            'MIME-Version' => '1.0',
+            'Content-Type' => 'text/plain; charset=UTF-8; format=flowed; delsp=yes',
+            'Content-Transfer-Encoding' => '8Bit',
+            'X-Mailer' => 'Drupal',
+            'Cc' => $cc,
+            'Bcc' => $bcc,
+        );
+        if (!drupal_mail('research_migration', 'research_migration_proposal_approved', $email_to, language_default(), $params, $from, true)) {
+            drupal_set_message('Error sending email message.', 'error');
+        }
+
+        drupal_set_message('esim research migration proposal No. ' . $proposal_id . ' approved. User has been notified of the approval.', 'status');
```

Comment:
Rephrase message as eSim Research Migration with Proposal No. <propsal_id> has been approved. The contributor has been notified of the approval